### ISSUE TITLE
Issues with "Handle Functional Event"

### ISSUE THREAD

**Issue by jakearchibald** [w3c/ServiceWorker#1196]
Sorry for the vague title, I couldn't explain it briefly.

https://w3c.github.io/ServiceWorker/#handle-functional-event.

"Handle Functional Event" wants an `ExtendableEvent`, a registration, and some callback steps. These callback steps are run on the active service worker's event loop. However, I need to be on the service worker's event loop in order to create the `ExtendableEvent`.

Maybe the algorithm should take the event name instead of an `ExtendableEvent`, and we just expect other specs to call "Update Service Worker Extended Events Set" themselves.

Also, it feels like "Update Service Worker Extended Events Set" only needs to take an `ExtendableEvent`, since the active worker could be determined from the global the `ExtendableEvent` was created in.

So other specs would do:

1. [Handle a functional event](https://w3c.github.io/ServiceWorker/#handle-functional-event) for *registration* and `"eventName"` by running the following steps:
  1. Let *event* be a trusted event (…new event object).
  1. Dispatch *event* on the [current global object](https://html.spec.whatwg.org/multipage/webappapis.html#current-global-object).
  1. [Update Service Worker Extended Events Set](https://w3c.github.io/ServiceWorker/#update-service-worker-extended-events-set) with *event*.

**Comment by jungkees** [comment]
> "Handle Functional Event" wants an ExtendableEvent, a registration, and some callback steps. These callback steps are run on the active service worker's event loop. However, I need to be on the service worker's event loop in order to create the ExtendableEvent.

Right. I checked two call sites, Notifications API and Push API. Notifications API was creating an event within the callbackSteps without giving an event argument to Handle Funcational Event. (So, the arguments and the algorithm input params needed to be clarified.) Push API was creating an event from the caller's steps. Yeah, so we need to clarify this. I left review comments in #1199.

**Comment by monica-ch** [comment]
Bug triage: Follow Jake's suggestion https://github.com/w3c/ServiceWorker/issues/1196#issuecomment-331160864

**Comment by yoshisatoyanagisawa** [comment]
I think #1813 is the final piece for the issue. Let me close the issue. Please feel free to reopen if not.
